pkg/util/yaml/decoder.go

```diff
@@ -226,9 +242,13 @@ func (d *YAMLOrJSONDecoder) Decode(into interface{}) error {
 
 			start := strings.LastIndex(js[:syntax.Offset], "\n") + 1
 			line := strings.Count(js[:start], "\n")
-			return fmt.Errorf("json: line %d: %s", line, syntax.Error())
+			return JSONSyntaxError{fmt.Errorf("json: line %d: %s", line, syntax.Error())}
```

Comment:
can we preserve the line and error data as fields in the JSONSyntaxError, and stringify it in JSONSyntaxError#Error?